staging/src/k8s.io/apimachinery/pkg/util/proxy/websocket_server.go

```diff
@@ -0,0 +1,207 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package proxy
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/util/httpstream/wsstream"
+	"k8s.io/apimachinery/pkg/util/remotecommand"
+	"k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/klog/v2"
+)
+
+const defaultIdleConnectionTimeout = 4 * time.Hour
+
+// Options contains details about which streams are required for
+// remote command execution.
+type options struct {
+	stdin  bool
+	stdout bool
+	stderr bool
+	tty    bool
+}
+
+// NewOptions creates a new Options from the Request.
+func streamOptionsFromRequest(req *http.Request) (*options, error) {
+	query := req.URL.Query()
+	tty := query.Get("tty") == "true"
+	stdin := query.Get("stdin") == "true"
+	stdout := query.Get("stdout") == "true"
+	stderr := query.Get("stderr") == "true"
```

Comment:
check how we were doing before, because it was working with "1" instead of "true" https://github.com/kubernetes/kubernetes/pull/119186/commits/91efc088394da4508fe255df93a451832c8eef2a#r1262726603

Reply:
agreed, don't re-parse from the request in ways that differ from our existing parameter parsing, pass in the stdin/stdout/stderr/tty options we already have.

ExecREST#Connect and AttachREST#Connect already have this info parsed into PodAttachOptions / PodExecOptions instances you can use when constructing the NewStreamTranslatorHandler